lib/ansible/plugins/inventory/scaleway.py

```diff
@@ -86,28 +86,44 @@
 
 from ansible.errors import AnsibleError
 from ansible.plugins.inventory import BaseInventoryPlugin, Constructable
-from ansible.module_utils.scaleway import SCALEWAY_LOCATION
+from ansible.module_utils.scaleway import SCALEWAY_LOCATION, parse_pagination_link
 from ansible.module_utils.urls import open_url
 from ansible.module_utils._text import to_native
 
+import ansible.module_utils.six.moves.urllib.parse as urllib_parse
 
-def _fetch_information(token, url):
-    try:
-        response = open_url(url,
-                            headers={'X-Auth-Token': token,
-                                     'Content-type': 'application/json'})
-    except Exception as e:
-        raise AnsibleError("Error while fetching %s: %s" % (url, to_native(e)))
-
-    try:
-        raw_json = json.loads(response.read())
-    except ValueError:
-        raise AnsibleError("Incorrect JSON payload")
 
-    try:
-        return raw_json["servers"]
-    except KeyError:
-        raise AnsibleError("Incorrect format from the Scaleway API response")
+def _fetch_information(token, url):
+    last = False
+    results = []
+    paginated_url = url
+    while not last:
+        try:
+            response = open_url(paginated_url,
+                                headers={'X-Auth-Token': token,
+                                         'Content-type': 'application/json'})
+        except Exception as e:
+            raise AnsibleError("Error while fetching %s: %s" % (url, to_native(e)))
+        try:
+            raw_json = json.loads(response.read())
+        except ValueError:
+            raise AnsibleError("Incorrect JSON payload")
+
+        try:
+            results.extend(raw_json["servers"])
+        except KeyError:
+            raise AnsibleError("Incorrect format from the Scaleway API response")
+
+        link = response.getheader('Link')
+        if link:
```

Comment:
```
if not link:
    return results
```

Reply:
I can move return inside the loop (I'll have to use it twice), but that also means deleting the loop logic since it has been made redundant. Up to you if you're ok with infinite loops (I can't really tell how long I'll iterate unless I do parse the pagination URLs to extract parameters). It looks solid from my point of view, should return or error in one way or another, but it could be a coding convention problem.